cmd/survival/main.go

```diff
@@ -0,0 +1,275 @@
+					delete(m.incantate, i)
+					m.point++
+
+					m.spells = reOrder(m.spells, spellMap[i])
+				}
+			}
+			if m.invoked[0] == i {
+				break
+			}
+			m.invoked[1] = m.invoked[0]
+			m.invoked[0] = i
+		case "d":
+			// Invoke here to prevent unnecessary reordering of spells every time a user triggers
+			// a spell casting.
+			if _, ok := m.incantate[m.invoked[0]]; ok {
+				delete(m.incantate, m.invoked[0])
+				m.point++
+				m.timer.Timeout += 2 * time.Second
+
+				m.spells = reOrder(m.spells, spellMap[m.invoked[0]])
+			}
+		case "f":
+			if _, ok := m.incantate[m.invoked[1]]; ok {
+				delete(m.incantate, m.invoked[1])
+				m.point++
+				m.timer.Timeout += 2 * time.Second
+
+				m.spells = reOrder(m.spells, spellMap[m.invoked[0]])
+			}
+		}
+	case timer.TickMsg:
+		var cmd tea.Cmd
+		m.timer, cmd = m.timer.Update(msg)
+		return m, cmd
+	case timer.TimeoutMsg:
+		m.isQuit = true
+		return m, tea.Quit
+	}
+
+	if len(m.incantate) == 0 {
+		if !m.cast {
+			m.cast, m.incantate, m.spells = gen(m.incantate)
+		} else {
+			m.cast, m.incantate, m.spells = gen(nil)
+		}
+	}
+	return m, nil
+}
+
+// TODO allocate slice
+// use strings builder
+// nice box
+// !!! center div Kappa
+func (m model) View() string {
+	var s string
+
+	if !m.cast {
+		s += "INVOKE\n\n"
+	} else {
+		s += "INVOKE AND CAST\n\n"
+	}
+
+	s += "| "
+	s += strings.Join(m.spells, " | ")
+	s += " |"
+	s += "\n--------------------------------------------------\n"
+
+	s += "| "
+	s += strings.Join(m.orbs, " | ")
+	s += " |"
+	s += "\n--------------------------------------------------\n"
+
+	s += "| Q | W | E |"
+	for _, i := range m.invoked {
+		s += " "
+		s += spellMap[i]
+		s += " |"
+	}
+	s += "\n--------------------------------------------------\n"
+
+	s += fmt.Sprintf("%d POINTS\n\n", m.point)
+	s += m.timer.View()
+	return s
+}
+
+func invoke(orbs []string) spell {
+	if len(orbs) < 3 {
+		return Undefined
+	}
+
+	// NOTE
+	// need to assign initial value to 1 (default 0)
+	// because in below we need do operation using multiplication
+	// to determine what type of spell
+	var invokerOrb rune = 1
+
+	orb := strings.Join(orbs, "")
+
+	for _, o := range orb {
+		invokerOrb *= o
+	}
+
+	return spellValue[invokerOrb]
+}
+
+// reOrder
+// Reordering the spells that will be displayed for the user to cast or invoke.
+// This reordering will be triggered if the user invokes or casts the correct answer.
+// The time complexity for this operation will be O(N).
+func reOrder(spells []string, answer string) []string {
+	newSpells := make([]string, 0, len(spells)-1)
+
+	for _, spell := range spells {
+		if spell != answer {
+			newSpells = append(newSpells, spell)
+		}
+	}
+
+	return newSpells
+}
+
+// TODO performance?
+// create random array of spells to incantate
+// it can either be 2 or 3 spells
+// if previous incantate is nil, create a new one with no overlap
+// else overlap is fine
+func gen(prev incantate) (bool, incantate, []string) {
+	// var cast bool
+	cast := true
```

Comment:
`cast` can be false/true in this mode

Reply:
I already make the `cast` can be true or false